tests/bugs/distribute/issue-2317.t

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+. $(dirname $0)/../../include.rc
+. $(dirname $0)/../../volume.rc
+
+TESTS_EXPECTED_IN_LOOP=126
+
+cleanup
+
+TEST glusterd
+TEST ${CLI} volume create ${V0} replica 3 ${H0}:/$B0/${V0}_{0..2}
+TEST ${CLI} volume start ${V0}
+
+TEST ${GFS} --volfile-server ${H0} --volfile-id ${V0} ${M0}
+
+# Create several files to make sure that at least some of them should be
+# migrated by rebalance.
+for i in {0..63}; do
```

Comment:
NIT: add a case of a hole at the end of a file.
E.g.: `truncate -s 1M <file>`

Reply:
This is a backport. If you think it's an interesting case to test, I can create a new patch for devel branch.

Reply:
Ok, did not notice it is already merged in the devel br. 
And from another look at dht_rebalance_sparse_segment() the case always covered :)
/* After a data segment there should always be a hole, since EOF is considered a hole. */